### PR TITLE
feat: ✅ added linting

### PR DESCRIPTION
**Pull request by abhagsain**
## Problem / Change

I have added `eslint`, `prettier`, `husky` and `lint-staged`  so that we format and fix any errors before committing our code. 

> Please note that the pre-commit hook will only run on the changes that are staged or getting committed.

## Screenshots (if any)

Here's an example

image

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+node_modules
+public/**
+.next/**
+.husky/**
```
Comment (dewanshparashar) [inline .eslintignore:4]: End of file.

File: .husky/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+_
```
Comment (dewanshparashar) [inline .husky/.gitignore:1]: Why is `_` added here? Pls, add comments.

File: .prettierignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+node_modules
+package-lock.json
```
Comment (dewanshparashar) [inline .prettierignore:2]: End of file

File: package.json [hunk 4]
```diff
@@ -5,18 +5,37 @@
...
+    "typescript": "4.5.5",
+    "husky": "^7.0.0"
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,json,ts,tsx}": [
+      "eslint",
+      "prettier --write"
+    ]
   }
 }
```
Comment (dewanshparashar) [inline package.json:41]: Any reason we have got yarn.lock instead of package-lock.json?

**Review by dewanshparashar — CHANGES_REQUESTED** [review]
Some minor comments identified.

File: package.json (hunk 4 above)
Comment (abhagsain) [reply package.json:41]: Because this project was built with Yarn that's why we have `yarn.lock`.

**Review by dewanshparashar — APPROVED** [review]
LGTM. Approved.

**Pull request merged**